### PR TITLE
feat(TDKN-246): Add `daikon-spring-aks-compat` module

### PR DESCRIPTION
**Pull request by fhuaulme**
**What is the problem this Pull Request is trying to solve?**

See https://jira.talendforge.org/browse/TDKN-246

TL;DR; Provide helper configuration in case AWS libraries are loaded in an Azure context.

**What is the chosen solution to this problem?**
 
* Add new configuration for AKS compatibility mode when AWS libraries are loaded.
* Add documentation for module usage.
* Add configuration test to module.

**Link to the JIRA issue**
https://jira.talendforge.org/browse/TDKN-246
 
**Please check if the Pull Request fulfills these requirements**
- [x] The PR commit message follows our [guidelines](https://github.com/Talend/daikon/blob/master/CONTRIBUTING.md)
- [ ] Tests for the changes have been added (for bug fixes / features, coverage should be over 75% in the new code)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Related design / discussions / pages (not in JIRA), if any, are all linked or available in the Pull Request

### CODE REVIEW

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/daikon_pull-request_master/2/) [:star:](https://ci-common.datapwn.com/job/daikon_pull-request_master/2/)

**Comment by sgandon** [comment]
I am not a big fan of the "aks" naming put everywhere, cause this would also be utsefull for any non aws provider.
